src/transformStateWithClones.js

```diff
@@ -7,7 +7,56 @@
  * @return {Object[]}
  */
 function transformStateWithClones(state, actions) {
-  // write code here
+  let obj = { ...state };
+  const res = [];
+
+  for (let i = 0; i < actions.length; i++) {
+    switch (actions[i].type) {
+      case 'addProperties':
+        obj = {
+          ...obj,
+          ...actions[i].extraData,
+        };
+        res.push({ ...obj });
```

Comment:
[DONT REPEAT YOURSELF] - If you perform same action in all switch cases - do it just once afterwards.